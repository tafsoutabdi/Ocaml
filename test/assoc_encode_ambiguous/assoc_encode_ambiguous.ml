open Tokenization.Assoc.Tokenizer
open Utils.Assoc
open Utils.Std

let print_context voc s = 
  Format.printf "voc = %s@.s = \"%s\"@.encode voc s = "
    (string_of_vocabulary voc) s

let () = 
  let voc = [("ca", 13); ("can", 7); ("ada", 12); ("nada", 72)] in
  let s = "canada" in
  print_context voc s;
  print_id_list (encode voc s);
  Format.printf "@."

  let () = 
  let voc = [("c", 1); ("can", 7); ("canal", 92); ("ada", 12); ("ca", 13); ("nada", 72)] in
  let s = "canada" in
  print_context voc s;
  print_id_list (encode voc s);
  Format.printf "@."
