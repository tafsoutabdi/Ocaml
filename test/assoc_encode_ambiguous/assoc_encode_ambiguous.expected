voc = [("ca", 13); ("can", 7); ("ada", 12); ("nada", 72)]
s = "canada"
encode voc s = [7; 12]
voc = [("c", 1); ("can", 7); ("canal", 92); ("ada", 12); ("ca", 13); ("nada", 72)]
s = "canada"
encode voc s = [7; 12]
